R/join-by.R

```diff
@@ -5,58 +5,61 @@
 #' `by` argument to any of the join functions (such as [left_join()]).
 #'
 #' # Join types
-#' `join_by()` supports four types of joins, as described below:
 #'
-#' * Equi joins
-#' * Non-equi joins
-#' * Rolling joins
-#' * Overlap joins
+#' Joins can be structured into two categories:
+#' - Equality joins
+#' - Non-equi joins, an umbrella term that contains: cross joins, inequality
+#'   joins, rolling joins, and overlap joins.
 #'
-#' ## Equi joins
+#' Equality, inequality, rolling, and overlap joins are discussed in more detail
+#' below. Cross joins are implemented through `cross_join()`.
```

Comment:
```suggestion
#' below. Cross joins are implemented through [cross_join()].
```

TODO: Waiting on `cross_join()` PR